addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/meter/DSMRMeter.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.internal.meter;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.openhab.binding.dsmr.internal.device.cosem.CosemObject;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemObjectType;
+import org.openhab.binding.dsmr.internal.device.cosem.OBISIdentifier;
+import org.openhab.binding.dsmr.internal.discovery.DSMRMeterDetector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * DSMR Meter represents a meter for this binding.
+ *
+ * Since the DSMR specification has evolved over time in combination we would
+ * help the OH2 user to easily add things the DSMR meter are auto discovered
+ * {@link DSMRMeterDetector}.
+ * To be able to discover meters and handle a correct administration of detected meters
+ * the DSMRMeter class has the following helper classes:
+ * - {@link DSMRMeterType} describing the meter type following a certain DSMR specification
+ * and provides the information which of the {@link CosemObject} that are part of the P1 telegram
+ * are supported for this meter
+ * - {@link DSMRMeterKind} describing what kind of meter (electricity, gas, etc.) this is. This information
+ * is mainly needed for the autodiscovery proces since a set of {@link CosemObject} could fit multiple meter types
+ * (i.e. Electricity for DSMR can have meter type ELECTRICITY_V4 and ELECTRICITY_V404. The meter kind for both types
+ * is ELECTRICITY. The auto discovery proces knows know both types describe the same kind and can find the most
+ * appropriate one.
+ * - {@link DSMRMeterIdentification} describing the identification of this meter.
+ *
+ * {@link DSMRMeterType} and {@link DSMRMeterIdentification} are private members of the DSMRMeter
```

Comment:
Is this javadoc comment still up-to-date. It suggest there are private members in this class that are not here.